doc/statistics/univariate_statistics.qbk

```diff
@@ -252,6 +263,38 @@ You should have /very/ good cause to pass negative values to the Gini coefficien
 Another use case is found in signal processing, but the sorting is by magnitude and hence has a different implementation.
 See `absolute_gini_coefficient` for details.
 
+[heading Mode]
+
+Compute the mode(s) of a data set:
+
+    std::vector<int> v {1, 3, 2, 2, 5, 4};
+    std::vector<int> modes; 
+    boost::math::statistics::mode(v, std::back_inserter(modes));
+    // Mode is 2, modes.size() == 1
+    std::deque<int> d_modes;
+    std::array<int, 7> w {2, 2, 3, 1, 5, 4, 4};
+    boost::math::statistics::mode(w, std::back_inserter(d_modes));
+    // Modes are 2 and 4, d_modes.size() == 2
+
+/Nota bene/: The input data is altered: in particular, it is sorted. Makes a call to `std::sort`, and as such requires random access iterators.
+
+If your data is sorted, the following function can be used instead:
+
+    std::vector<int> v {1, 2, 2, 3, 4, 5};
+    std::vector<int> modes; 
+    boost::math::statistics::sorted_mode(v, std::back_inserter(modes));
+    // Mode is 2, modes.size() == 1
+    std::deque<int> d_modes;
+    std::array<int, 7> w {1, 2, 2, 3, 4, 4, 5};
+    boost::math::statistics::sorted_mode(w, std::back_inserter(d_modes));
+    // Modes are 2 and 4, d_modes.size() == 2  
+
+/Nota bene/: The requirements for sorted_mode are reduced to forward iterators because there is no call to `std::sort`.
+
+/Nota bene/: Passing unsorted data to sorted_mode is a bug.
```

Comment:
That's not strictly true, right? It doesn't have to be sorted, it just has to have equal values be consecutive, which is the same requirement as for `std::unique`. Sorting is just the simplest way to achieve it.

Reply:
@jeremy-murphy You are correct. Consecutive equal values would work here.

Reply:
I would probably describe this in a literary sense as a "distracting caveat"; not a true statements about the code need to be stated.

But I think having use cases would clarify whether this needs to be explicitly stated; do we have any? I personally have never needed to compute a mode.

Reply:
@NAThompson I think of a case where data being handled by sorted_mode would be configured differently than what you would get from `std::sort`